### PR TITLE
postgresql17: new port

### PR DESCRIPTION
**Pull request by dgilman**
#### Description

I expect this to build!

It must be merged after https://github.com/macports/macports-ports/pull/26042

Also see: https://github.com/macports/macports-ports/pull/24825

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 14.6.1 23G93 x86_64
Xcode 16.0 16A242d

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by dgilman** [comment]
This will probably fail as the `postgresql_select-17` port is not yet available

File: databases/postgresql17/Portfile [hunk 1]
```diff
@@ -0,0 +1,179 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           compilers 1.0
+PortGroup           compiler_blacklist_versions 1.0
+PortGroup           muniversal 1.0
+PortGroup           legacysupport 1.1
+
+name                postgresql17
+version             17.0
```
Comment (reneeotten) [inline databases/postgresql17/Portfile:10]: looks like upstream just released version `17.1` - @dgilman can you update to that version please? I'll merge the PR once that has happened. Thanks!

**Comment by ecbrown** [comment]
I'm interested in this port, as well.  I looked through the port files and noticed:

```
conflicts           postgresql96-doc postgresql10-doc postgresql11-doc postgresql12-doc \
    postgresql13-doc postgresql14-doc postgresql15-doc
```

and I wonder if this should also include postgresql16-doc

File: databases/postgresql17/Portfile (hunk 1 above)
Comment (dgilman) [reply databases/postgresql17/Portfile:10]: I will bump it tomorrow

**Comment by holymonson** [comment]
@dgilman Hi, PostgreSQL 18 is out, are you willing to make a new port?

**Comment by dgilman** [comment]
I opened the PR for postgresql 18 already. It is awaiting review

**Pull request merged**